common/djangoapps/microsite_configuration/__init__.py

```diff
@@ -1 +1,19 @@
+from django.conf import settings as base_settings
+
+from microsite_configuration import microsite
 from .templatetags.microsite import page_title_breadcrumbs
+
+
+class MicrositeAwareSettings():
+    """
+    This class is a handy utility to make a call to the settings
+    completely microsite aware by replacing the:
+    from django.conf import settings
+    with:
+    from microsite_configuration import settings
+    """
+
+    def __getattr__(self, name):
+        return microsite.get_value(name, base_settings.__getattr__(name))
+
+settings = MicrositeAwareSettings()
```

Comment:
The one thing that I feel is missing with this microsite aware settings object is the fact that it would allow a microsite to override ANY setting. That could open too much space for microsite to break things. I could add a filter to this in the form of a list of allowed_overrides. If a given name is not in this list then it would directly take the global setting for it.